### PR TITLE
[#21] Make the Platform Attributes validation check match Platform validation

### PR DESCRIPTION
**Pull request by cyrus-dev**
This pull request forces the supply chain validation service to verify that the Platform Credential has a status of PASS.  If it does not, no matter the outcome of the Attributes validation, the status of the Attributes can not be PASS.  This has to do with the attributes passing with no platform certificate.

Fix for #21

### CODE REVIEW

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java [hunk 1]
```diff
@@ -151,8 +153,24 @@ public SupplyChainValidationSummary validateSupplyChain(final EndorsementCredent
...
                     PlatformCredential pc = it.next();
-                    validations.add(validatePlatformCredentialAttributes(pc, device.getDeviceInfo(),
-                            ec));
+                    SupplyChainValidation attributeScv = validatePlatformCredentialAttributes(
+                            pc, device.getDeviceInfo(), ec);
+
+                    if (platformScv != null) {
+                        if (platformScv.getResult() == AppraisalStatus.Status.FAIL
+                                || platformScv.getResult() == AppraisalStatus.Status.ERROR) {
+                            if (attributeScv.getResult() ==  AppraisalStatus.Status.PASS) {
```
Comment (apldev3) [inline HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: Is this `attributeScv` guaranteed to be non-null if `platformScv` isn't?

**Comment by apldev3** [comment]
This looks good pending the outcome of the one question I have.

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java (hunk 1 above)
Comment (cyrus-dev) [reply HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: I'm not 100% certain, but I was just going back over this and from all indications validatePlaformCredentialAttributes via buildValidationRecord return the object SupplyChainValidation.  I'm fine either way.

Comment (apldev3) [reply HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: Actually, based on how this reads, I don't think this update will work in all instances. Having the single instance of `platformScv` won't work when we have multiple Platform Credentials to validate. The last PC provided by the Iterator in the first block will be the one with that sets `platformScv` object.

Instead, a mapping is needed from `PlatformCredential`s to `SupplyChainValidation`s. Then you as you iterate through the Platform Credential Attribute Supply Chain Validation loop that you have here, you can get the corresponding `scv` for the current Platform Credential and perform the check as you are.

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java [hunk 2]
```diff
@@ -151,8 +153,24 @@ public SupplyChainValidationSummary validateSupplyChain(final EndorsementCredent
                 Iterator<PlatformCredential> it = pcs.iterator();
                 while (it.hasNext()) {
                     PlatformCredential pc = it.next();
-                    validations.add(validatePlatformCredentialAttributes(pc, device.getDeviceInfo(),
-                            ec));
+                    SupplyChainValidation attributeScv = validatePlatformCredentialAttributes(
+                            pc, device.getDeviceInfo(), ec);
+
+                    if (platformScv != null) {
```
[the remark this answers is not in the document]
Comment (cyrus-dev) [reply HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: Removing check and based on the agreement, resolving the first comment.

[the remark this answers is not in the document]
Comment (apldev3) [reply HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: I suppose it doesn't hurt to have it then, but I'm fairly certain you can't enable PcAttributeValidation unless PcValidation has been enabled previously, at least insofar as using only the ACA Portal is concerned. The only way I could see triggering this is direct manipulation at the DB level. All that being said, it's cheap to check for nulls so we may as well do it.

**Comment by apldev3** [comment]
I checked `PlatformCredential` and it does have a functioning HashCode, so it shouldn't be an issue that we're throwing it in a `HashSet`. At this point, I'd say this is good to go in.

**Pull request merged**